File: 1500-count-largest-group/1500-count-largest-group.cs
public class Solution {
    public int CountLargestGroup(int n) {
        var sums = new int[37];

        int sumOfDig(int num) {
            return (num == 0 ? 0 : (num % 10) + sumOfDig(num / 10));
        }

        for (int i = 1; i <= n; ++i)
            sums[sumOfDig(i)]++;

        int ans = 0, max = 0;

        for (int i = 0; i < 37; i++)
            if (sums[i] > max) {
                max = sums[i]; 
                ans = 1;
            }
            else if (sums[i] == max)
                ++ans;        
        return ans;
    }
}

File: 1500-count-largest-group/1500-count-largest-group.go
func sumOfDig(num int) int {
    if num == 0 {
        return 0
    }
    return num % 10 + sumOfDig(num / 10)
}

func countLargestGroup(n int) int {
    sums := make([]int, 37)
    for i := 1; i <= n; i++ {
        sums[sumOfDig(i)]++
    }
    ans, max := 0, 0
    for i := 0; i < 37; i++ {
        if sums[i] > max {
            max = sums[i]
            ans = 1
        } else if sums[i] == max {
            ans += 1
        }
    }
    return ans
}